pkg/envoy/ads/cache_callbacks.go

```diff
@@ -44,18 +45,59 @@ func (s *Server) OnStreamOpen(ctx context.Context, streamID int64, typ string) e
 	}
 
 	s.proxyRegistry.RegisterProxy(proxy)
+	go func() {
```

Comment:
Does this make sense to abstract into a separate function like in the previous implementation?

Reply:
sorry i prematurely merged this.. i'm still investigating ways we can abstract this stuff more. I've taken a note to revisit this :)